cabal-install/tests/UnitTests/Distribution/Solver/Modular/Solver.hs

```diff
@@ -2422,6 +2521,36 @@ dbIssue3775 =
   , Right $ exAv "B" 2 [ExAny "A", ExAny "warp"]
   ]
 
+-- A database where the setup depends on something which has a test stanza, does the
+-- test stanza get enabled?
+dbSetupStanza :: ExampleDb
+dbSetupStanza =
+  [ Right $
+      exAv "A" 1 []
+        `withSetupDeps` [ExAny "B"]
+  , Right $
+      exAv "B" 1 []
+        `withTest` exTest "test" [ExAny "C"]
+  ]
+
+-- With the user constraint syntax (fails)
+setupStanzaTest1 :: SolverTest
+setupStanzaTest1 = userConstraints ["B test"] $ mkTest dbSetupStanza "setupStanzaTest1" ["A"] (solverSuccess [("A", 1), ("B", 1)])
```

Comment:
Indeed, I'm not sure. @mpickering may recall but I will otherwise drop that part of the comment. This test seems to be the same as `setupStanzaTest2` but use `userConstraints`